File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution(object):
    def maxOperations(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        cnt = 0
        i, j = 0, len(nums) - 1
        nums.sort()
        
        while i < j:
            sum = nums[i] + nums[j]
            if sum == k:
                cnt += 1
                i += 1
                j -= 1
            elif sum < k:
                i += 1
            else:
                j -= 1
                
            
        return cnt

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.rs
impl Solution {
    pub fn max_operations(mut nums: Vec<i32>, k: i32) -> i32 {
        nums.sort();
        
        let mut cnt = 0;
        let (mut i, mut j) = (0, nums.len() - 1);
        
        while i < j {
            let sum = nums[i] + nums[j];
            if sum == k {
                cnt += 1;
                i += 1;
                j -= 1;
            } else if sum < k {
                i += 1;
            } else {
                j -= 1;
            }
        }
        
        cnt
    }
}